Primo.js

```diff
@@ -314,12 +304,16 @@ function importPNX(text) {
 	}
 	
 	// the three letter ISO codes that should be in the language field work well:
-	item.language = ZU.xpathText(doc, '//display/language');
+	item.language = ZU.xpathText(doc, '(//display/language|//facets/language)[1]');
 	
 	var pages = ZU.xpathText(doc, '//display/format');
-	if(pages && pages.search(/[0-9]+/) != -1) {
-		pages = pages.replace(/[\(\)\[\]]/g, "").match(/[0-9]+/);
-		item.pages = item.numPages = pages[0];
+	if(pages && pages.search(/\d/) != -1) {
+		if (item.itemType == 'book') {
+			item.numPages = extractNumPages(pages)
+		} else {
+			pages = pages.replace(/[\(\)\[\]]/g, "").match(/[0-9]+/);
+			item.pages = item.numPages = pages[0];
+		}
```

Comment:
Do you have an example for the `else` part? (For articles there is also the utility function <a href="https://github.com/zotero/zotero/blob/4.0/chrome/content/zotero/xpcom/utilities.js#L644">getPageRange</a>, however the metadata for articles should be better handled with the addata part from below.)